### PR TITLE
Create details dialog for application exploring - Closes #4356, #4357

### LINKED ISSUE #4356: Create details dialog for application exploring
### Description

Create a component to explore the selected applications information and use this component in modal route

### Acceptance Criteria

- The component should have the following: avatar, custom background, application name, owner address, pin status, application status, chain id, last update time, Last Certificate Height, node url 
- Allow users to pin/unpin an application
- Component should use the custom background in header, if the background doesn’t exists show default background based on application configuration
- Ensure [UI/UX design](https://www.figma.com/file/uStjr4eQ37wPm9bDt8JyXP/Applications-management?node-id=2%3A1351) requirements are met
- Add required unit tests
- Add required e2e tests

### Additional Information

- Suggested component name `BlockchainApplicationDetails`
- Route: “blockchain/application/details?chainId=XXXX”
- Enable pin flag based on “usePinBlockchainApplication” hook

```
const {pins, setPin, deletePin} = usePinBlockchainApplication()
// Pins = [chainId]
```

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4356, #4357
### How was it solved?

- [x]  Create a component to have the following: avatar, custom background, application name, owner address, pin status, application status, chain id, last update time, Last Certificate Height, node url.
- [x] Allow the component to be accessible via the dialog
- [x] Add unit test
- [x] Add e2e test

### How was it tested?

- e2e
- jest

### CODE REVIEW

File: setup/jest.config.js [hunk 1]
```diff
@@ -52,7 +53,7 @@ module.exports = {
     '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
       '<rootDir>/tests/__mocks__/fileMock.js',
   },
-  collectCoverage: true,
+  collectCoverage: false,
```
Comment (soroushm) [inline setup/jest.config.js]: true

File: src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+  return (
+    <Dialog hasClose className={`${grid.row} ${grid['center-xs']}`}>
+      <div className={styles.wrapper}>
+        <div />
+        <div>
+          <div className={styles.avatarContainer}>
+            {/* just a place holder */}
+            <div>
+              sdf
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js]: what is the `sdf`?

File: src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      </span>,
+      content: <span className={`${styles.detailContentText} last-update`}>{moment(lastUpdated).format('DD MMM YYYY')}</span>,
+    },
+    {
+      header: <span className={styles.headerText}>
+        {t('Last Certificate Height')}
+      </span>,
+      content: <span className={`${styles.detailContentText} last-certificate-height`}>{lastCertificateHeight}</span>,
+    },
+  ];
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js]: there is no point to have an array of element! maybe we can have the value as array and use it

```suggestion
  const footerDetails = [
    {
      header: {
      title: t('Chain ID'),
      class:  [],
       tooltip: t(''), //needs meesages
       },
      value: chainId,
    },
    {
      header: {
        title: t('Status'),
        class: [styles.statusChip, styles[state]]
        },
        value: t(state)
  }
  ];
```

File: src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                // TODO: this is just a place holder link pending when its part of the response payload from service
+                to="https://enevti.com/"
+              >
+                <Icon name="chainLinkIcon" className={styles.hwWalletIcon} />
+                {t('https://enevti.com/')}
+              </Link>
+            </div>
+            <div className={styles.balanceRow}>
+              <span>Deposited:</span>
+              <span>5,351.859 LSK</span>
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js]: all the value should come from the variable like `data.deposit`

**Review by soroushm — COMMENTED** [review]
quick review

File: packages/common/store/reducers/index.js [hunk 5]
```diff
@@ -11,3 +11,4 @@ export { default as watchList } from '@dpos/validator/store/reducers/watchList';
 export { default as service } from './service';
 export { default as loading } from './loading';
 export { account } from 'src/modules/account/store/reducer';
+export { blockChainApplications } from 'src/modules/blockchainApplication/explore/store/reducer';
```
Comment (ikem-legend) [inline packages/common/store/reducers/index.js]: @eniolam1000752 This should be aliased along with the previous line

File: packages/views/screens/router/routesMap.js [hunk 6]
```diff
@@ -41,6 +41,7 @@ import SwitchAccount from '@account/components/SwitchAccount';
 import BackupRecoveryPhraseFlow from '@account/components/BackupRecoveryPhraseFlow';
 import RemoveCurrentAccountFlow from 'src/modules/account/components/RemoveCurrentAccountFlow';
 import RemoveSelectedAccountFlow from 'src/modules/account/components/RemoveSelectedAccountFlow';
+import BlockchainApplicationDetails from 'src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails';
```
Comment (ikem-legend) [inline packages/views/screens/router/routesMap.js]: @eniolam1000752 I believe these should be aliased too

File: src/modules/blockchainApplication/api/index.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * Retrieves the details of a single transaction
+ *
+ * @param {Object} data
+ * @param {String} data.params
+ * @param {String} data.params.id - Id of the transaction
+ * @param {String?} data.baseUrl - Lisk Service API url to override the
+ * existing ServiceUrl on the network param. We may use this to retrieve
+ * the details of an archived transaction.
+ * @param {Object} data.network - Network setting from Redux store
+ * @returns {Promise} Transaction details API call
```
Comment (ikem-legend) [inline src/modules/blockchainApplication/api/index.js]: @eniolam1000752 Please update the documentation

File: src/modules/blockchainApplication/explore/store/action.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import actionTypes from './actionTypes';
+
+/**
+ * Trigger this action to set blockchain application pin
+ *
+ * @returns {Object} - Action object
+ */
+export const pinApplication = (chainId) => ({
+  type: actionTypes.setApplicationPin,
+  chainId,
```
Comment (ikem-legend) [inline src/modules/blockchainApplication/explore/store/action.js]: @eniolam1000752 For consistency with the rest of the actions, data should be the key for the Redux action, same way type is always a key

File: src/modules/blockchainApplication/explore/store/action.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+});
+
+export const removePinnedApplication = (chainId) => ({
+  type: actionTypes.removeApplicationPin,
+  chainId,
```
Comment (ikem-legend) [inline src/modules/blockchainApplication/explore/store/action.js]: @eniolam1000752 Same applies here

File: src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+const application = {
+  data: {
+    name: 'Test app',
+    chainID: 'aq02qkbb35u4jdq8szo3pnsq',
+    state: 'active',
+    address: 'lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu',
+    lastCertificateHeight: 1000,
+    lastUpdated: 123456789,
+    deposit: 5e10,
+    serviceUrl: 'https://enevti.com/',
```
Comment (soroushm) [inline src/modules/blockchainApplication/explore/components/BlockchainApplicationDetails/BlockchainApplicationDetails.js]: is the actual response
{
  "data": [{
    "name": "Test app",
    "chainID": "aq02qkbb35u4jdq8szo3pnsq",
    "state": "active",
    "address": "lsk24cd35u4jdq8szo3pnsqe5dsxwrnazyqqqg5eu",
    "lastCertificateHeight": 1000, // Optional
    "lastUpdated": 123456789, // Optional
  }],
  "meta": {
    "count": 10,
    "offset": 10,
    "total": 150,
  },
  "links": {}
}

File: tests/constants/selectors.js [hunk 11]
```diff
@@ -293,6 +294,13 @@ const ss = {
   verifyPublicKeyInput: '.publicKey',
   verifySignatureInput: '.signature',
   acceptTermsButton: '.accept-terms',
+
```
Comment (ikem-legend) [inline tests/constants/selectors.js]: @eniolam1000752 This empty line should be removed

**Comment by ikem-legend** [comment]
@eniolam1000752 Please fix CSS for application status badge

**Review by ikem-legend — DISMISSED** [review]

**Review by soroushm — DISMISSED** [review]

**Review by ikem-legend — APPROVED** [review]

**Review by soroushm — APPROVED** [review]

**Pull request merged**